tests/network/salt_master.pm

```diff
@@ -0,0 +1,111 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016-2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test Salt stack on two machines. Here we test mainly the
+#  master but minion is also present just for having more of those.
+# Maintainer: Pavel Dostal <pdostal@suse.cz>
+
+use base "consoletest";
+use strict;
+use testapi;
+use lockapi;
+use mm_network;
+use utils qw(zypper_call pkcon_quit systemctl);
+use version_utils 'is_sle';
+use registration 'add_suseconnect_product';
+
+sub run {
+    my ($self) = @_;
+    my $hostname = get_var('HOSTNAME');
+    barrier_create('SALT_MINIONS_READY', 2);
+    barrier_create('SALT_FINISHED',      2);
+    select_console 'root-console';
+
+    # Configure static network, disable firewall
+    configure_default_gateway;
+    systemctl 'stop ' . $self->firewall;
+    configure_static_ip('10.0.2.101/24');
+    configure_static_dns(get_host_resolv_conf());
+
+    # Set the hostname to identify both minions
+    assert_script_run "hostnamectl set-hostname $hostname";
+    assert_script_run "hostnamectl status|grep $hostname";
+    assert_script_run "hostname|grep $hostname";
+
+    # Install both salt master and minion and run the master daemon
+    zypper_call("in salt-master salt-minion");
+    systemctl("start salt-master");
+    systemctl("status salt-master");
+
+    # Set the right address of the salt master and run the minion
+    assert_script_run('sed -i -e "s/#master: salt/master: 10.0.2.101/" /etc/salt/minion');
+    assert_script_run("cat /etc/salt/minion | grep 'master: '");
+    systemctl("start salt-minion");
+    systemctl("status salt-minion");
+
+    # before accepting the key, wait until the minion is fully started (systemd might be not reliable)
+    assert_script_run('salt-run state.event tagmatch="salt/auth" quiet=True count=1', timeout => 300);
+    barrier_wait 'SALT_MINIONS_READY';
+
+    # List and accept both minions when they are ready
+    assert_script_run("salt-key -L");
+    assert_script_run("salt-key --accept-all -y");
+    # try to ping the minion. If it does not respond on the first try the ping
+    # might have gone lost so try more often. Also see bsc#1069711
+    unless (script_run "salt '*' test.ping") {
```

Comment:
`salt '*' test.ping` does not return 1, will not work

https://github.com/saltstack/salt/issues/18510